plugins/inputs/prometheus/consul.go

```diff
@@ -0,0 +1,174 @@
+package prometheus
+
+import (
+	"bytes"
+	"context"
+	"log"
+	"net/url"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/hashicorp/consul/api"
+)
+
+type ConsulServiceInfo struct {
+	ID             string
+	Node           string
+	Address        string
+	Datacenter     string
+	NodeMeta       map[string]string
+	ServiceID      string
+	ServiceName    string
+	ServiceAddress string
+	ServiceTags    []string
+	ServiceMeta    map[string]string
+	ServicePort    int
+}
+
+func convertConsulService(s *api.CatalogService) *ConsulServiceInfo {
+	return &ConsulServiceInfo{
+		ID:             s.ID,
+		Node:           s.Node,
+		Address:        s.Address,
+		Datacenter:     s.Datacenter,
+		NodeMeta:       s.NodeMeta,
+		ServiceID:      s.ServiceID,
+		ServiceAddress: s.ServiceAddress,
+		ServiceTags:    s.ServiceTags,
+		ServiceMeta:    s.ServiceMeta,
+		ServicePort:    s.ServicePort,
+	}
+}
```

Comment:
Is there a reason not to use `api.CatalogService` directly? Your declaration is a one-to-one translation...

Reply:
Probably not, originally I wanted to limit the number of fields that can be used in the template, but that might not be necessary.